Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -232,6 +232,26 @@ struct SymbolGraphLoader {
             for (selector, _) in symbol.mixins {
                 if var symbolAvailability = (symbol.mixins[selector]?["availability"] as? SymbolGraph.Symbol.Availability) {
                     guard !symbolAvailability.availability.isEmpty else { continue }
+                    // For platforms with a fallback option (e.g., Catalyst and iOS), apply the explicit availability annotation of the fallback platform when it is not explicitly available on the primary platform.
+                    DefaultAvailability.fallbackPlatforms.forEach { (fallbackPlatform, inheritedPlatform) in
+                        let fallbackAvailability = symbolAvailability.availability.first(where: {
+                            $0.domain?.rawValue == fallbackPlatform.rawValue
+                        })
+                        let inheritedAvailability = symbolAvailability.availability.first(where: {
+                            $0.domain?.rawValue == inheritedPlatform.rawValue
+                        })
+                        let defaultAvailability = defaultAvailabilities.first(where: { $0.platformName ==  fallbackPlatform })
+                        if var inheritedAvailability, let fallbackAvailability, let defaultAvailability {
+                            // Ensure that the availability version is not overwritten if the symbol has an explicit availability annotation for that platform.
+                            if SymbolGraph.SemanticVersion(string: defaultAvailability.introducedVersion!)?.description == fallbackAvailability.introducedVersion?.description {
```

Comment:
Is it safe to force-unwrap here? Wouldn't that crash if the platform was declared as explicitly unavailable in the Info.plist?

Reply:
Also, AFAICT `SemanticVersion` is equatable, so it shouldn't be necessity to compare its string descriptions.